### PR TITLE
Drop old Symfony code

### PR DESCRIPTION
**Pull request by jordisala1991**
I am targeting this branch, because this is BC.

### Subject

Removing old code that never gets executed because we dropped SF 2.3 to 2.7

### CODE REVIEW

File: src/Resources/stubs/symfony2.php [hunk 1]
```diff
@@ -1,21 +0,0 @@
-<?php
```
Comment (jordisala1991) [inline src/Resources/stubs/symfony2.php]: I don't know why this class was there

File: tests/Block/AbstractBlockServiceTest.php [hunk 2]
```diff
@@ -1,23 +0,0 @@
-<?php
```
Comment (jordisala1991) [inline tests/Block/AbstractBlockServiceTest.php]: Am I allowed to remove classes from tests directory?

File: tests/Block/AbstractBlockServiceTest.php [hunk 3]
```diff
@@ -1,23 +0,0 @@
...
-use Sonata\BlockBundle\Test\AbstractBlockServiceTestCase;
-
-/**
- * NEXT_MAJOR: remove this class.
- *
- * @deprecated Deprecated since version 3.1. Use Sonata\BlockBundle\Test\AbstractBlockServiceTestCase instead
- */
-abstract class AbstractBlockServiceTest extends AbstractBlockServiceTestCase
-{
-}
```
Comment (greg0ire) [inline tests/Block/AbstractBlockServiceTest.php]: Are you sure this is supposed to be removed?

Comment (jordisala1991) [reply tests/Block/AbstractBlockServiceTest.php]: The correct file is on src directory, but I am not sure If I am allowed to remove it from here. 

One thing is for sure: https://github.com/sonata-project/SonataBlockBundle/blob/3.x/composer.json#L47

Comment (greg0ire) [reply tests/Block/AbstractBlockServiceTest.php]: Oh right I didn't notice that.

File: src/Resources/stubs/symfony2.php (hunk 1 above)
Comment (jordisala1991) [reply src/Resources/stubs/symfony2.php]: Restored, this is necessary

**Comment by greg0ire** [comment]
Thanks @jordisala1991 !

File: src/Block/Service/MenuBlockService.php [hunk 4]
```diff
@@ -163,19 +164,12 @@ protected function getFormSettingsKeys()
...
-
-        // NEXT_MAJOR: remove SF 2.7+ BC
-        if (method_exists('Symfony\Component\Form\AbstractType', 'configureOptions')) {
-            $choices = array_flip($choices);
-
-            // choice_as_value options is not needed in SF 3.0+
-            if (method_exists('Symfony\Component\Form\FormTypeInterface', 'setDefaultOptions')) {
-                $choiceOptions['choices_as_values'] = true;
-            }
+        // choice_as_value options is not needed in SF 3.0+
```
Comment (OskarStark) [inline src/Block/Service/MenuBlockService.php]: should we name it with `NEXT_MAJOR` ?

Comment (jordisala1991) [reply src/Block/Service/MenuBlockService.php]: Probably yes, to be honest when I search for code removal I usually start with method_exists class_exists and interface_exists, then 5.3 and finally NEXT_MAJOR

**Pull request merged**